test/manifolds/segre.jl

```diff
@@ -0,0 +1,4149 @@
+using Manifolds, Test
+
+# Approximate derivative of f at x
+# https://en.wikipedia.org/wiki/Finite_difference_coefficient
+function finite_difference(f::Function, x::Float64, h::Float64; order=1::Int64) #={{{=#
+    if order == 1
+        return (
+            (1 / 12) * f(x - 2 * h) +
+            (-2 / 3) * f(x - 1 * h) +
+            (2 / 3) * f(x + 1 * h) +
+            (-1 / 12) * f(x + 2 * h)
+        ) / h
+    elseif order == 2
+        return (
+            (-1 / 12) * f(x - 2 * h) +
+            (4 / 3) * f(x - 1 * h) +
+            (-5 / 2) * f(x) +
+            (4 / 3) * f(x + 1 * h) +
+            (-1 / 12) * f(x + 2 * h)
+        ) / h^2
+    elseif order == 3
+        return (
+            (1 / 8) * f(x - 3 * h) +
+            (-1) * f(x - 2 * h) +
+            (13 / 8) * f(x - 1 * h) +
+            (-13 / 8) * f(x + 1 * h) +
+            (1) * f(x + 2 * h) +
+            (-1 / 8) * f(x + 3 * h)
+        ) / h^3
+    end
+end #=}}}=#
+
+# Manifolds to test
+Ms = [
```

Comment:
For sure, we could test only one Segre with each "valence". Then we could for example test
```
    Segre{ℝ,(10,)}(),
    Segre{ℝ,(7, 2)}(),
    Segre{ℝ,(7, 9, 9)}(),
    Segre{ℝ,(9, 3, 6, 6)}(),
    MetricManifold(Segre{ℝ,(10,)}(), WarpedMetric{1.2025837056880606}()),
    MetricManifold(Segre{ℝ,(2, 9)}(), WarpedMetric{1.1302422072971439}()),
    MetricManifold(Segre{ℝ,(9, 6, 10)}(), WarpedMetric{1.4545138169484464}()),
    MetricManifold(Segre{ℝ,(9, 3, 8, 10)}(), WarpedMetric{1.396673190458706}()),
```

Reply:
That already looks better, I would still change them to `Segre(10)`, `Segre(7,2)`, and so on, since that is much more readable.
Since these are all real – is it planned (or covered even in theory) to do complex Segre ones?